File: backend/app/schemas/branch_manager.py
from datetime import datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field


class ApplicationDecisionRequest(BaseModel):
    # CLAUDE.md §26: "escalate" is the only action available once an
    # application is over the branch's delegated limit — enforced
    # server-side (an over-limit "approve" is rejected, never trusted from
    # the client).
    decision: Literal["approve", "reject", "escalate"]
    comments: str = Field(min_length=1)


class BranchQueueItemResponse(BaseModel):
    id: int
    customer_full_name: str
    customer_email: str
    loan_product_name: str
    amount_requested: Decimal
    created_at: datetime
    # CLAUDE.md §26: min(branch override, product default) — shown so the
    # UI can disable "Approve"/"Reject" and only offer "Escalate" when true,
    # while the server enforces the same rule independently.
    effective_limit: Decimal
    over_limit: bool

File: frontend/src/schemas/branchManager.ts
import { z } from "zod";

// Mirrors backend app/schemas/branch_manager.py
export const applicationDecisionSchema = z.object({
  decision: z.enum(["approve", "reject", "escalate"]),
  comments: z.string().min(1, "Comments are required"),
});
export type ApplicationDecisionInput = z.infer<typeof applicationDecisionSchema>;

export interface BranchQueueItemResponse {
  id: number;
  customer_full_name: string;
  customer_email: string;
  loan_product_name: string;
  amount_requested: string;
  created_at: string;
  effective_limit: string;
  over_limit: boolean;
}
